### PR TITLE
configury: fix pmix/ext2x support

### PR DESCRIPTION
**Pull request by ggouaillardet**
- fix pmix/ext2x support
 - fix hwloc < 2.0 support
 - fix macro handling
 - remove unused files

### CODE REVIEW

File: opal/hwloc/hwloc-internal.h [hunk 1]
```diff
@@ -190,8 +192,6 @@ OPAL_DECLSPEC extern bool opal_hwloc_use_hwthreads_as_cpus;
 #define HWLOC_OBJ_L3CACHE HWLOC_OBJ_CACHE
 #define HWLOC_OBJ_L2CACHE HWLOC_OBJ_CACHE
 #define HWLOC_OBJ_L1CACHE HWLOC_OBJ_CACHE
-#define HWLOC_OBJ_NUMANODE HWLOC_OBJ_NODE
-#define HWLOC_OBJ_PACKAGE HWLOC_OBJ_SOCKET
```
Comment (rhc54) [inline opal/hwloc/hwloc-internal.h]: I believe you will still need these as hwloc 1.11.x doesn't know the numanode and package values

**Comment by rhc54** [comment]
I'm surprised as I also have a centos 7 box and it works fine with the distributed version of hwloc v1.11 - how did you build your version of it?

**Comment by ggouaillardet** [comment]
I just found these macros are defined in `hwloc`'s `deprecated.h` if 1.10 <= hwloc < 2.0,
and pushed an updated commit.

I configure'd with ```--enable-picky --prefix=... CFLAGS='-g -O0 -pipe'```

**Comment by rhc54** [comment]
FWIW: my default centos 7 version is hwloc 1.11.2 - and things work fine with it. Let me know what you find.

**Comment by ggouaillardet** [comment]
This is the error message I have without the patch, I will investigate "stock" `hwloc 1.11.2` tomorrow

```
$ make hwloc/hwloc.lo
  CC       hwloc/hwloc.lo
In file included from ../../../src/pprte/opal/hwloc/hwloc.c:25:0:
../../../src/pprte/opal/hwloc/hwloc-internal.h:195:28: error: 'HWLOC_OBJ_NODE' undeclared here (not in a function)
 #define HWLOC_OBJ_NUMANODE HWLOC_OBJ_NODE
                            ^
/home/gilles/local/hwloc-v1.11/include/hwloc/deprecated.h:27:24: note: in expansion of macro 'HWLOC_OBJ_NUMANODE'
 #define HWLOC_OBJ_NODE HWLOC_OBJ_NUMANODE
                        ^
../../../src/pprte/opal/hwloc/hwloc.c:42:5: note: in expansion of macro 'HWLOC_OBJ_NODE'
     HWLOC_OBJ_NODE,
     ^
../../../src/pprte/opal/hwloc/hwloc-internal.h:196:27: error: 'HWLOC_OBJ_SOCKET' undeclared here (not in a function)
 #define HWLOC_OBJ_PACKAGE HWLOC_OBJ_SOCKET
                           ^
/home/gilles/local/hwloc-v1.11/include/hwloc/deprecated.h:25:26: note: in expansion of macro 'HWLOC_OBJ_PACKAGE'
 #define HWLOC_OBJ_SOCKET HWLOC_OBJ_PACKAGE
                          ^
../../../src/pprte/opal/hwloc/hwloc.c:43:5: note: in expansion of macro 'HWLOC_OBJ_SOCKET'
     HWLOC_OBJ_SOCKET,
     ^
make: *** [hwloc/hwloc.lo] Error 1
```

**Comment by rhc54** [comment]
I think I've figured out what is going on. I had the "stock" hwloc installed, but not hwloc-devel, and so it was picking up my v2.0 headers from my local build. With the v1.11.2 hwloc-devel installed, I still don't get your errors, but I do hit the following (minus any of these patches):
```shell
../../../opal/.libs/libprteopen-pal.so: undefined reference to `hwloc_distances_release'
../../../opal/.libs/libprteopen-pal.so: undefined reference to `hwloc_shmem_topology_adopt'
../../../opal/.libs/libprteopen-pal.so: undefined reference to `hwloc_topology_set_io_types_filter'
../../../opal/.libs/libprteopen-pal.so: undefined reference to `hwloc_distances_get_by_depth'
```
I'll take these patches and see where that leads.

**Comment by rhc54** [comment]
I'm finding a configure issue that I really don't understand. The configure logic is adding flags to point to my local software installation of libevent, which is fine. However, the PMIx ext3x component is failing because it insists that the hwloc.la file _also_ be installed in that location. This happens even when I add --with-hwloc=foo --with-hwloc-lib=foo/lib64 to the configure line. 

Not sure why these two are being tied together. Perhaps a second pair of eyes?

**Comment by ggouaillardet** [comment]
Can you post your full `configure` command and error message ?
Did you build PMIx with hwloc support ? Is it the same hwloc in `foo` ?
Can you post your PMIx configure command and `libpmix.la` file too ?

**Comment by rhc54** [comment]
Oh my - you caught me! I failed to rebuild PMIx and so it does indeed point to the other hwloc.

Sigh - let me try fixing that and report back. I suspect you hit it right on the head 😄

**Comment by rhc54** [comment]
Yep - that fixed it. My bad - thanks for the second pair of eyes!!

**Pull request merged**